miner/src/miner.rs

```diff
@@ -94,7 +94,8 @@ impl Miner {
                     Ok((pow_hash, work, nonce)) => {
                         self.submit_nonce(pow_hash, work, nonce);
                         if self.limit != 0 && self.nonces_found >= self.limit {
-                            break;
+                            debug!("miner nonce limit reached, terminate ...");
+                            std::process::exit(0);
```

Comment:
I just replaced `std::process::exit(0)` by `broadcast_exit_signals()` on my local environmentr, miner will exit successfully when `limit` reached.

Reply:
Adding a cli `bats` test for the miner's `--limit` flag would be an improvement, and it can be done by the example of `ckb run`:
https://github.com/nervosnetwork/ckb/blob/fc9d3e6f17547c0d183c8d1cdfcca56a031c7d08/util/app-config/src/tests/graceful_shutdown.bats#L5-L17